open-source-hosts-plugin/scripts/AwsTools/SumerianHostScript.ts

```diff
@@ -0,0 +1,207 @@
+      (animGroup) => animGroup.name === config.bindPoseOffsetName
+    );
+
+    const animClips = await HostObject.loadCharacterAnimations(
+      this.getScene(),
+      this,
+      bindPoseOffset,
+      config.animClipPaths
+    );
+    const gestureConfig = await HostObject.loadJson(config.gestureConfigPath);
+
+    // set up animations
+    const assets = {
+      characterMesh: this,
+      animClips,
+      gestureConfig,
+      poiConfig,
+      bindPoseOffset,
+    };
+    this.host = HostObject.assembleHost(assets, this.getScene());
+  }
+
+  protected initPointOfInterestTracking(poiConfig, lookJoint): void {
+    // have the Host track the scene's active camera
+    HostObject.addPointOfInterestTracking(
+      this.host,
+      this._scene,
+      poiConfig,
+      lookJoint
+    );
+
+    // Track the active camera by default.
+    const camera = this.getScene().activeCamera;
+    this.host.PointOfInterestFeature.setTarget(camera);
+  }
+
+  protected async initTextToSpeech(): Promise<void> {
+    const region = this.awsConnector.getRegion();
+    const credentials = this.awsConnector.getCredentials();
+
+    // setting this global config is necessary -
+    // else the Polly SDK's first call to Cognito.getID(...) will fail
+    AWS.config.region = region;
+
+    const customUserAgent = `AWSToolsForBabylonJSEditor-${PLUGIN_VERSION}`;
+
+    const pollyClient = new AWS.Polly({credentials, region, customUserAgent});
+    const pollyPresigner = new AWS.Polly.Presigner({
+      service: pollyClient,
+    });
+
+    await HostObject.initTextToSpeech(pollyClient, pollyPresigner);
+
+    HostObject.addTextToSpeech(
+      this.host,
+      this.getScene(),
+      this.pollyVoiceId,
+      this.pollyEngine,
+      this.pollyLanguageId
+    );
+  }
+
+  /**
+   * Call this method to signal when the host has finished its asynchronous
+   * initialization and is ready for use.
+   */
+  protected onHostReady(): void {
+    this.onHostReadyObserver.notifyObservers(this);
+  }
+
+  /**
+   * If the scene is being run from the editor, this method will return the version of
+   * the configuration that uses local paths.
+   * Otherwise, this method will return the version of the configuration that uses paths
+   * relative to the workspace directory.
+   * @returns {SumerianHostMetadata}
+   */
+  protected getMetadata(): SumerianHostMetadata {
+    if (this.isRunFromEditor()) {
+      return this.metadata.editor.sumerian;
+    }
+
+    return this.metadata.sumerian;
+  }
+
+  /**
+   * NodeJS defines the global variable 'process' -- as does recent versions of
+   * FireFox, which is why we check the name as well.
+   * We use this to determine whether the code is being run locally from the editor,
+   * or is running in the browser
+   * @returns boolean
+   */
+  protected isRunFromEditor(): boolean {
+    return (
+      typeof process !== 'undefined' &&
+      process &&
+      process.release.name === 'node'
+    );
+  }
```

Comment:
This sounds more like 'isRunFromElectron' Right? If someone distributes an electron app that uses babylonjs/hosts, will this return true?

Reply:
This is code predates this PR. I think you may be right. Our objective here is to determine whether the host script is executing in the context of the Editor's design-time viewport or whether it's executing in an actual full runtime context. I looked into alternatives briefly but didn't find any non-brittle way to make this determination. I'd like to stick with this existing implementation for this PR, but I'll create a new backlog item to look into a more robust solution in the future. Acceptable?

Reply:
yep!